packages/firestore/src/local/local_store.ts

```diff
@@ -979,12 +1114,59 @@ export class LocalStore {
   }
 }
 
+export function newLocalStore(
+  /** Manages our in-memory or durable persistence. */
+  persistence: Persistence,
+  queryEngine: QueryEngine,
+  initialUser: User
+): LocalStore {
+  return new LocalStoreImpl(persistence, queryEngine, initialUser);
+}
+
+/**
+ * An interface on top of LocalStore that provides additional functionality
+ * for MultiTabSyncEngine.
+ */
+export interface MultiTabLocalStore extends LocalStore {
+  /** Returns the local view of the documents affected by a mutation batch. */
+  lookupMutationDocuments(batchId: BatchId): Promise<MaybeDocumentMap | null>;
+
+  removeCachedMutationBatchMetadata(batchId: BatchId): void;
+
+  setNetworkEnabled(networkEnabled: boolean): void;
+
+  getActiveClients(): Promise<ClientId[]>;
+
+  getTarget(targetId: TargetId): Promise<Target | null>;
+
+  /**
+   * Returns the set of documents that have been updated since the last call.
+   * If this is the first call, returns the set of changes since client
+   * initialization. Further invocations will return document changes since
+   * the point of rejection.
+   */
+  getNewDocumentChanges(): Promise<MaybeDocumentMap>;
+
+  /**
+   * Reads the newest document change from persistence and forwards the internal
+   * synchronization marker so that calls to `getNewDocumentChanges()`
+   * only return changes that happened after client initialization.
+   */
+  synchronizeLastDocumentChangeReadTime(): Promise<void>;
+}
+
 /**
  * An implementation of LocalStore that provides additional functionality
  * for MultiTabSyncEngine.
+ *
+ * Note: some field defined in this class might have public access level, but
+ * the class is not exported so they are only accessible from this file. This is
```

Comment:
s/file/module

Please also remove duplicate comments for `MultiTabLocalStoreImpl`.

Reply:
Done.